lib/src/agent.cc

```diff
@@ -63,13 +63,17 @@ void Agent::start() {
         } while (true);
 
         // The agent is now connected and the request handlers are
-        // set; we can now call the monitoring method that will block
+        // set; we can now call the monitoring method and then block
         // this thread of execution.
         // Note that, in case the underlying connection drops, the
         // connector will keep trying to re-establish it indefinitely
         // (the max_connect_attempts is 0 by default).
         LOG_DEBUG("PCP connection established; about to start monitoring it");
-        connector_ptr_->monitorConnection();
+        connector_ptr_->startMonitoring();
+        pcp_util::chrono::seconds pause { 1 };
+
+        while (true)
+            pcp_util::this_thread::sleep_for(pause);
```

Comment:
As a side note, the agent was born like that: https://github.com/puppetlabs/pxp-agent/blob/34ea1d01ec449bbeb524f7d4a8e46b21c22fde55/src/agent.cpp#L166-L168